diff --git a/node_modules/html2canvas/dist/lib/render/background.js b/node_modules/html2canvas/dist/lib/render/background.js
index 3491637..699fe66 100644
--- a/node_modules/html2canvas/dist/lib/render/background.js
+++ b/node_modules/html2canvas/dist/lib/render/background.js
@@ -27,14 +27,16 @@ var calculateBackgroundPaintingArea = function (backgroundClip, element) {
 };
 exports.calculateBackgroundPaintingArea = calculateBackgroundPaintingArea;
 var calculateBackgroundRendering = function (container, index, intrinsicSize) {
-    var backgroundPositioningArea = exports.calculateBackgroundPositioningArea(exports.getBackgroundValueForIndex(container.styles.backgroundOrigin, index), container);
-    var backgroundPaintingArea = exports.calculateBackgroundPaintingArea(exports.getBackgroundValueForIndex(container.styles.backgroundClip, index), container);
-    var backgroundImageSize = exports.calculateBackgroundSize(exports.getBackgroundValueForIndex(container.styles.backgroundSize, index), intrinsicSize, backgroundPositioningArea);
+    var backgroundPositioningArea = calculateBackgroundPositioningArea(getBackgroundValueForIndex(container.styles.backgroundOrigin, index), container);
+    var backgroundPaintingArea = calculateBackgroundPaintingArea(getBackgroundValueForIndex(container.styles.backgroundClip, index), container);
+    var backgroundImageSize = calculateBackgroundSize(getBackgroundValueForIndex(container.styles.backgroundSize, index), intrinsicSize, backgroundPositioningArea);
     var sizeWidth = backgroundImageSize[0], sizeHeight = backgroundImageSize[1];
-    var position = length_percentage_1.getAbsoluteValueForTuple(exports.getBackgroundValueForIndex(container.styles.backgroundPosition, index), backgroundPositioningArea.width - sizeWidth, backgroundPositioningArea.height - sizeHeight);
-    var path = exports.calculateBackgroundRepeatPath(exports.getBackgroundValueForIndex(container.styles.backgroundRepeat, index), position, backgroundImageSize, backgroundPositioningArea, backgroundPaintingArea);
+    var position = getAbsoluteValueForTuple(getBackgroundValueForIndex(container.styles.backgroundPosition, index), backgroundPositioningArea.width - sizeWidth, backgroundPositioningArea.height - sizeHeight);
+    var path = calculateBackgroundRepeatPath(getBackgroundValueForIndex(container.styles.backgroundRepeat, index), position, backgroundImageSize, backgroundPositioningArea, backgroundPaintingArea);
     var offsetX = Math.round(backgroundPositioningArea.left + position[0]);
     var offsetY = Math.round(backgroundPositioningArea.top + position[1]);
+    sizeWidth = Math.max(1, sizeWidth);
+    sizeHeight = Math.max(1, sizeHeight);
     return [path, offsetX, offsetY, sizeWidth, sizeHeight];
 };
 exports.calculateBackgroundRendering = calculateBackgroundRendering;
